ystia/yorc
# Feature/google block storage

# Pull Request description

**Support GCE Block Storages**

As: an app manager
I want to: attach block storage on an GCE Compute
So that: I can persist informations on those persistent storages

AC1: Could provision a new volume and attach it to a compute
AC2: Could reuse an existing volume and attach it to a compute
AC3: Compute could have multiple volumes attached
AC4: All above should work with scalable computes

## Description of the change

### What I did
- Create yorc.nodes.google.PersistentDisk node type
- Create relationship **yorc.relationships.google.AttachesTo** with specific prop **mode** to allow
the possibility for a compute to attach a disk with **read only** mode.
- Create **scratch_disks** prop for google compute instance to attach local SSD disks.

### How I did it
- Use the existing code around topology modification for BlockStorage and compute **local_storage** 
requirement.

### How to verify it
- Simple topology with a scalable compute and one or many block storages.
You can use **yorc.relationships.google.AttachesTo** and set mode property to **READ_ONLY**
You can use generic **tosca.relationships.AttachTo**.
You can attach existing disk by setting **volume_id** property.
Check the deployment and the resources creation on gcloud console


## Applicable Issues
fixes #81
